client/trino-client/src/main/java/io/trino/client/ResultRows.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.client;
+
+import java.util.Iterator;
+import java.util.List;
+
+import static java.util.Collections.emptyIterator;
+
+/**
+ * Represents row-wise iterable result set data. This interface is meant to be extended in the future
+ * when other encoding formats are introduced.
+ */
+public interface ResultRows
+        extends Iterable<List<Object>>
+{
+    ResultRows NULL_ROWS = new ResultRows() {
+        @Override
+        public boolean hasNoRows()
+        {
+            // This should be the only instance of this method returning true,
+            // as this means "no rows yet" which is different from "empty rows".
+            return true;
+        }
+
+        @Override
+        public Iterator<List<Object>> iterator()
+        {
+            return emptyIterator();
+        }
+    };
+
+    static ResultRows fromIterableRows(Iterable<List<Object>> values)
+    {
+        return values::iterator;
+    }
+
+    default boolean hasNoRows()
```

Comment:
just kee `hasRows`. The names are similar and having two methods actually makes code harder to read.

Reply:
I don't like inverted method calls